spring-cloud-task-docs/src/main/asciidoc/batch-starter.adoc

```diff
@@ -0,0 +1,532 @@
+and other file formats. In order to read from a file, you can provide some components
+yourself via normal Spring configuration (`LineTokenizer`, `RecordSeparatorPolicy`,
+`FieldSetMapper`, `LineMapper`, or `SkippedLinesCallback`). You can also use the
+following properties to configure the reader as required.
+
+.`FlatFileItemReader` Properties
+|===
+| Property | Type | Default Value | Description
+
+| `spring.batch.job.flatfileitemreader.saveState`
+| `boolean`
+| `true`
+| Determines if the state should be saved for restarts
+
+| `spring.batch.job.flatfileitemreader.name`
+| `String`
+| `null`
+| Name used to provide unique keys in the `ExecutionContext`
+
+| `spring.batch.job.flatfileitemreader.maxItemcount`
+| `int`
+| `Integer.MAX_VALUE`
+| Maximum number of items to be read from the file
+
+| `spring.batch.job.flatfileitemreader.currentItemCount`
+| `int`
+| 0
+| Number of items that have already been read. Used on restarts.
+
+| `spring.batch.job.flatfileitemreader.comments`
+| `List<String>`
+| empty List
+| A list of Strings that indicate commented lines (lines to be ignored) in the file.
+
+| `spring.batch.job.flatfileitemreader.resource`
+| `Resource`
+| `null`
+| The resource to be read.
+
+| `spring.batch.job.flatfileitemreader.strict`
+| `boolean`
+| `true`
+| If set to true, the reader will throw an exception if the resource is not found.
+
+| `spring.batch.job.flatfileitemreader.encoding`
+| `String`
+| `FlatFileItemReader.DEFAULT_CHARSET`
+| Encoding to be used when reading the file.
+
+| `spring.batch.job.flatfileitemreader.linesToSkip`
+| `int`
+| 0
+| Indicates the number of lines to skip at the start of a file.
+
+| `spring.batch.job.flatfileitemreader.delimited`
+| `boolean`
+| `false`
+| Indicates if the file is a delimited file (CSV, etc). Only this property or `spring.batch.job.flatfileitemreader.fixedLength` can be `true` at the same time.
+
+| `spring.batch.job.flatfileitemreader.delimiter`
+| `String`
+| `DelimitedLineTokenizer.DELIMITER_COMMA`
+| If reading a delimited file, indicates the delimiter to parse on.
+
+| `spring.batch.job.flatfileitemreader.quoteCharacter`
+| `char`
+| `DelimitedLineTokenizer.DEFAULT_QUOTE_CHARACTER`
+| Used to determine what character is used to quote values.
+
+| `spring.batch.job.flatfileitemreader.includedFields`
+| `List<Integer>`
+| empty list
+| A list of indicies of which fields in a record to include in the item.
+
+| `spring.batch.job.flatfileitemreader.fixedLength`
+| `boolean`
+| `false`
+| Indicates if a file's records are parsed via column numbers. Only this property or `spring.batch.job.flatfileitemreader.delimited` can be `true` at the same time.
+
+| `spring.batch.job.flatfileitemreader.ranges`
+| `List<Range>`
+| empty list
+| List of column ranges to parse a fixed width record by. Read about Range https://docs.spring.io/spring-batch/docs/4.3.x/api/org/springframework/batch/item/file/transform/Range.html[here]
```

Comment:
end with period.

Reply:
Change to:

List of column ranges by which to parse a fixed width record. See the documentation about Range https://docs.spring.io/spring-batch/docs/4.3.x/api/org/springframework/batch/item/file/transform/Range.html[Range].